benchmarks/seq2seq/benchmark.py

```diff
@@ -0,0 +1,130 @@
+        """Returns the number of vocabularies, including UNK and EOS."""
+
+        return self._vocab_size
+
+    def _generate_sentence(self, xp):
+        """Returns the dummy sentence.
+
+        The returned value is a single ``xp.ndarray``.
+        """
+
+        # +2 for UNK and EOS
+        sentence = xp.arange(self._words_per_data, dtype=xp.int32) + 2
+        xp.random.shuffle(sentence)
+        return sentence
+
+    def generate_training_data(self, xp):
+        """Returns the dummy data for training.
+
+        The returned value is a list of instances.
+        A instance is a tuple of two ``xp.ndarray``\\ s which represents
+        a set of source sentence and target sentence.
+        """
+
+        return [(self._generate_sentence(xp), self._generate_sentence(xp))
+                 for _ in range(self._count)]
+
+    def generate_validation_data(self, xp):
+        """Returns the dummy data for validation.
+
+        The returned value is a list of ``xp.ndarray``, which represents
+        a input sentence.
+        """
+
+        return [self._generate_sentence(xp) for _ in range(self._count)]
+
+
+def _convert(batch, device):
+    # Arrays in the batch are already on the target device.
+    return {'xs': [x for x, _ in batch], 'ys': [y for _, y in batch]}
+
+
+class Application(object):
+
+    def __init__(self, unit=1024, layer=3, dummy_data=None):
+        # Setup dummy data generator
+        if dummy_data is None:
+            dummy_data = {}
+        generator = DummyDataGenerator(**dummy_data)
+        vocab_size = generator.get_vocab_size()
+
+        # Setup model
+        self.model = Seq2seq(layer, vocab_size, vocab_size, unit)
+        self.gpu = -1
+        if is_backend_gpu():
+            self.gpu = 0
+            chainer.cuda.get_device_from_id(self.gpu).use()
+            self.model.to_gpu()
+        elif is_backend_ideep():
+            self.model.to_intel64()
+
+        # Generate dummy data for training and validation
+        xp = self.model.xp
+        self.train_data = generator.generate_training_data(xp)
+        self.validate_data = generator.generate_validation_data(xp)
+
+    def train(self, batchsize=64, epoch=3):
+        # Setup optimizer
+        optimizer = chainer.optimizers.Adam()
+        optimizer.setup(self.model)
+
+        # Setup iterator
+        train_iter = chainer.iterators.SerialIterator(
+            self.train_data, batchsize)
+
+        # Setup updater and trainer
+        updater = training.updaters.StandardUpdater(
+            train_iter, optimizer, converter=_convert, device=self.gpu)
+        trainer = training.Trainer(updater, (epoch, 'epoch'))
+
+        # Add extensions (for debugging)
+        trainer.extend(training.extensions.LogReport(
+            trigger=(1, 'epoch')))
+        trainer.extend(training.extensions.PrintReport(
+            ['epoch', 'iteration', 'main/loss', 'main/perp', 'elapsed_time']),
+            trigger=(1, 'epoch'))
```

Comment:
these lines seem to should be executed in `__init__()`.

Reply:
Ok, moved them to  ``__init__``.